mail-filter/sqlgrey/sqlgrey-1.8.0.ebuild

```diff
@@ -0,0 +1,252 @@
+		local SQLgrey_DB_USER_PWD=""
+		for (( n=1; n<=16; n++ )); do
+			SQLgrey_DB_USER_PWD="${SQLgrey_DB_USER_PWD}${SQLgrey_PWD_MATRIX:$(($RANDOM%${#SQLgrey_PWD_MATRIX})):1}"
+		done
+	else
+		einfo "Reusing current database user password..."
+		local SQLgrey_DB_USER_PWD="${SQLgrey_CONF_PWD}"
+	fi
+	einfo
+
+	# Configure the various database backends
+	local SQLgrey_KEY_INPUT=""
+	einfo "Creating SQLgrey database backend data and configuration"
+	einfo "  Please select what kind of database you like to use:"
+	if use postgres || has_version dev-perl/DBD-Pg ; then
+		einfo "    [p] PostgreSQL"
+		SQLgrey_KEY_INPUT="${SQLgrey_KEY_INPUT},p"
+	fi
+	if use mysql || has_version dev-perl/DBD-mysql ; then
+		einfo "    [m] MySQL"
+		SQLgrey_KEY_INPUT="${SQLgrey_KEY_INPUT},m"
+	fi
+	if use sqlite || has_version dev-perl/DBD-SQLite ; then
+		einfo "    [s] SQLite"
+		SQLgrey_KEY_INPUT="${SQLgrey_KEY_INPUT},s"
+	fi
+	einfo "    [x] Exit"
+	SQLgrey_KEY_INPUT="${SQLgrey_KEY_INPUT},x"
+	einfo
+	einfo "  Press one of the keys [${SQLgrey_KEY_INPUT:1}]: "
+	while true; do
+		read -n 1 -s SQLgrey_DB_TYPE
+		case "${SQLgrey_DB_TYPE}" in
+			"p" | "P" )
+				SQLgrey_DB_TYPE="p"
+				einfo "  PostgreSQL database backend"
+				einfo
+				break
+			;;
+			"m" | "M" )
+				SQLgrey_DB_TYPE="m"
+				einfo "  MySQL database backend"
+				einfo
+				break
+			;;
+			"s" | "S" )
+				SQLgrey_DB_TYPE="s"
+				einfo "  SQLite database backend"
+				einfo
+				break
+			;;
+			"x" | "X" )
+				return 0
+			;;
+		esac
+	done
+
+	# If we don't use SQLite, the password must not be set
+	if [[ ( "${SQLgrey_DB_TYPE}" != "s" ) && ( "${SQLgrey_CONF_PWD}" != "" ) ]]; then
+		ewarn "This configuration is only for new installations. You seem to"
+		ewarn "have already a modified sqlgrey.conf"
+		ewarn "Do you want to continue?"
+		SQLgrey_KEY_INPUT="y,n"
+		einfo "   Press one of the keys [$SQLgrey_KEY_INPUT]: "
+		while true; do
+			read -n 1 -s SQLgrey_Ignore
+			case "$SQLgrey_Ignore" in
+				"y"|"Y" )
+					break
+				;;
+				"n"|"N" )
+					return 0
+				;;
+			esac
+		done
+	fi
+
+	## Per-RDBMS configuration ##
+	# POSTGRESQL
+	if [[ "${SQLgrey_DB_TYPE}" == "p" ]] ; then
+
+		ewarn "If prompted for a password, please enter your PgSQL postgres password"
+		ewarn ""
+
+		einfo "Creating SQLgrey PostgreSQL database \"${SQLgrey_DB_NAME}\" and user \"${SQLgrey_DB_USER_NAME}\""
+		"${ROOT}/usr/bin/psql" -h ${SQLgrey_DB_HOST} -d template1 -U postgres -c "CREATE USER ${SQLgrey_DB_USER_NAME} WITH PASSWORD '${SQLgrey_DB_USER_PWD}' NOCREATEDB NOCREATEUSER; CREATE DATABASE ${SQLgrey_DB_NAME}; GRANT ALL PRIVILEGES ON DATABASE ${SQLgrey_DB_NAME} TO ${SQLgrey_DB_USER_NAME}; GRANT ALL PRIVILEGES ON SCHEMA public TO ${SQLgrey_DB_USER_NAME}; UPDATE pg_database SET datdba=(SELECT usesysid FROM pg_shadow WHERE usename='${SQLgrey_DB_USER_NAME}') WHERE datname='${SQLgrey_DB_NAME}';" || die "psql db creation command failed"
+
+		einfo "Changing SQLgrey configuration in sqlgrey.conf"
+		sed -i \
```

Comment:
Can't you just put db type in a variable, and combine those `sed`s into one below ifs?